### ISSUE TITLE
doesn't work promise of init method

### ISSUE THREAD

**Issue by hbakhtiyor** [simonlast/node-persist#46]
doesn't handle onSuccess callback, also doesn't log none of them from the snippet of code

```js
storage.init({logging: true}).then(function() {
    console.log("loaded");
}, function(err) {
    console.log("error");
});
```

**Comment by akhoury** [comment]
@simonlast i think one is worth a patch release. if you want I can do that, but you need to add me on npm as well.
```
cd node-persist
npm owner add akhoury 

# i think
```
src: https://docs.npmjs.com/cli/owner

**Comment by akhoury** [comment]
this issue was occurring with already-created directories, the loading was happening but the promise was never resolved :/

**Comment by simonlast** [comment]
Added you!

**Comment by akhoury** [comment]
related: thanks @hbakhtiyor  for catching that. 
https://github.com/simonlast/node-persist/commit/508ab143a84d23fda70b6df45432da5134d84871

also published 0.0.8 - this project needs some [tests](https://github.com/simonlast/node-persist/issues/47)

**Comment by hbakhtiyor** [comment]
still reproducing at latest version

**Comment by akhoury** [comment]
I just tested again and I cannot reproduce this issue.

```javascript
var storage = require('node-persist');
storage.init({logging: true}).then(function() {
    console.log("loaded");
}, function(err) {
    console.log("error");
});
```

also, the [tests](https://github.com/simonlast/node-persist/blob/master/tests/index.js#L47-L50) include `init` testing with a promise, what happens when you run 

```
npm test
```

**Comment by hbakhtiyor** [comment]
i downloaded from npm registry, maybe published wrong version?

**Comment by akhoury** [comment]
nope - can you paste your full test case here please and the full log you're seeing in the results.

**Comment by hbakhtiyor** [comment]
my current ts code
 
```ts
    private initializeLocalStorage(): Promise<void> {
        return this.localStorage
            .init({
                continuous: false,
                logging: this.logger.info
            });
            .then(() => {
                this.logger.debug("successfully initialized local storage persister");
            })
            .catch((err: any) => {
                this.logger.error("error during initializing local storage persister", { error: err });
            });
    }
```

**Comment by akhoury** [comment]
and the error ?

**Comment by hbakhtiyor** [comment]
nothing error, just none of them (resolve & reject) have been called

**Comment by akhoury** [comment]
sorry can't reproduce - can you write test case in ES5 and with the native `console.log` that doesn't work.

My test cases works just fine.

**Comment by hbakhtiyor** [comment]
hmm, tested via node REPL, works fine
i think problem with my code

btw, 
`getItem` doesn't return promise
`setItem` doesn't invoke resolve & reject if `continuous` set to `false`

**Comment by hbakhtiyor** [comment]
@akhoury btw, the `init` promise works incorrectly, will load `key: value` data after resolving the promise

e.g.
```ts
get(key: string): Promise<any> {
    return this.initializeLocalStorage()
        .then(() => {
            const value = this.localStorage.getItem(key);
            this.logger.debug("successfully get value from local storage", { key: key, value: value });

            return value;
        });
}

private initializeLocalStorage(): Promise<void> {
    return this.localStorage
        .init({
            continuous: false,
            logging: this.logger.info
        })
        .then(() => {
            this.logger.debug("successfully initialized local storage persister");
        })
        .catch((err: any) => {
            this.logger.error("error during initializing local storage persister", { error: err });
        });
}
```
output
```
debug: successfully initialized local storage persister
debug: successfully get headers from local storage key=bar, value=undefined
info: loaded: /path/bar
....
```

the `value=undefined`, because the data not have been loaded, but promise was resolved

**Comment by akhoury** [comment]
@hbakhtiyor thanks for your feedback

> `getItem` doesn't return promise 

it synchronous because it's always reading from memory, we could still make it return a promise but that won't be backward compatible, see [this](https://github.com/simonlast/node-persist#getitemkey-callback---returns-value-synchronous-but-may-linger-async-call-if-unless-ttl-expired) - i do have plans to make it return a promise, maybe on the next major release, when i can break deprecated functions.

> `setItem` doesn't invoke resolve & reject if `continuous` set to `false`

Good catch, I opened a ticket for that https://github.com/simonlast/node-persist/issues/52 the promise shouldn't resolve immediately, only after a manual call to `storage.persist/persistSync` but yea, still a bug because that didn't happen when I tested it.

> the `init` promise works incorrectly, will load key: value data after resolving the promise

Also another good catch, here's its ticket, https://github.com/simonlast/node-persist/issues/53

i'm gonna close this one so we can follow up on others individually.
